### PR TITLE
Fixed the odo storage mount by adding the namespacedOpenshiftObject

### PR DESCRIPTION
**Pull request by mik-dass**
Issue: #534 
Signed-off-by: mdas mrinald7@gmail.com

### CODE REVIEW

**Comment by surajnarwade** [comment]
Current PVC name is `mystorage-myapp-pvc`, can we do this `mystorage-<component_name>-myapp-pvc`

**Comment by kadel** [comment]
@mik-dass can you please add mount/unmount to e2e tests?

**Comment by mik-dass** [comment]
@kadel Ok will try but in a seperate PR

**Comment by kadel** [comment]
> @kadel Ok will try but in a seperate PR

It should be in this one, to make sure that it was fixed.

File: tests/e2e/e2e_test.go [hunk 1]
```diff
@@ -354,6 +354,46 @@ var _ = Describe("odoe2e", func() {
...
+
+			It("should be able to unmount the storage using the storage name", func() {
+				runCmd("odo storage unmount pv2")
+
+				// Verify with deploymentconfig
+				getDc := runCmd("oc get dc/php-" + appTestName + " -o go-template='" +
+					"{{range .spec.template.spec.containers}}" +
+					"{{range .volumeMounts}}{{.name}}{{end}}{{end}}'")
+
+				Expect(getDc).NotTo(ContainSubstring("pvc2"))
```
Comment (kadel) [inline tests/e2e/e2e_test.go]: shouldn't it be `pv2`?

File: tests/e2e/e2e_test.go [hunk 2]
```diff
@@ -354,6 +354,46 @@ var _ = Describe("odoe2e", func() {
...
+
+			It("should be able to mount the storage to the path specified", func() {
+				runCmd("odo storage mount pv2 --path /mnt/pv2")
+
+				// Verify with deploymentconfig
+				getDc := runCmd("oc get dc/php-" + appTestName + " -o go-template='" +
+					"{{range .spec.template.spec.containers}}" +
+					"{{range .volumeMounts}}{{.name}}{{end}}{{end}}'")
+
+				Expect(getDc).To(ContainSubstring("pvc2"))
```
Comment (kadel) [inline tests/e2e/e2e_test.go]: shouldn't it be `pv2`?

File: tests/e2e/e2e_test.go [hunk 3]
```diff
@@ -354,6 +354,46 @@ var _ = Describe("odoe2e", func() {
...
+
+			It("should be able to unmount the storage", func() {
+				runCmd("odo storage unmount /mnt/pv2")
+
+				// Verify with deploymentconfig
+				getDc := runCmd("oc get dc/php-" + appTestName + " -o go-template='" +
+					"{{range .spec.template.spec.containers}}" +
+					"{{range .volumeMounts}}{{.name}}{{end}}{{end}}'")
+
+				Expect(getDc).NotTo(ContainSubstring("pvc2"))
```
Comment (kadel) [inline tests/e2e/e2e_test.go]: again, `pv2`?

File: tests/e2e/e2e_test.go (hunk 1 above)
Comment (mik-dass) [reply tests/e2e/e2e_test.go]: Oops that's why the test failed, fixing it

File: tests/e2e/e2e_test.go (hunk 2 above)
Comment (mik-dass) [reply tests/e2e/e2e_test.go]: Oops that's why the test failed, fixing it

File: tests/e2e/e2e_test.go (hunk 3 above)
Comment (mik-dass) [reply tests/e2e/e2e_test.go]: Oops that's why the test failed, fixing it

**Pull request merged**